src/libraries/Microsoft.PowerFx.Connectors/Execution/HttpFunctionInvoker.cs

```diff
@@ -63,50 +59,97 @@ public HttpRequestMessage BuildRequest(FormulaValue[] args)
             // Header names are not case sensitive.
             // From RFC 2616 - "Hypertext Transfer Protocol -- HTTP/1.1", Section 4.2, "Message Headers"
             var headers = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            string body = null;
 
             var map = _argMapper.ConvertToSwagger(args);
-            foreach (var param in _argMapper._openApiParameters)
+            foreach (var param in _argMapper.OpenApiParameters)
             {
-                if (map.TryGetValue(param.Name, out var paramValue))
+                if (param.In == null && param.Schema.Properties != null && param.Schema.Properties.Any())
+                {
+                    body = GetBody(param, map);
+                }
+                else if (map.TryGetValue(param.Name, out var paramValue))
                 {
                     var valueStr = paramValue.ToObject().ToString();
 
-                    switch (param.In.Value)
+                    if (param.In == null)
                     {
-                        case ParameterLocation.Path:
-                            path = path.Replace("{" + param.Name + "}", HttpUtility.UrlEncode(valueStr));
-                            break;
-
-                        case ParameterLocation.Query:
-                            query.Append((query.Length == 0) ? "?" : "&");
-                            query.Append(param.Name);
-                            query.Append('=');
-                            query.Append(HttpUtility.UrlEncode(valueStr));
-                            break;
-
-                        case ParameterLocation.Header:
-                            headers.Add(param.Name, valueStr);
-                            break;
-
-                        default:
-                            throw new NotImplementedException($"{param.In}");
+                        // No schema, we send the raw value
+                        body = paramValue.ToObject().ToString();
                     }
+                    else
+                    {
+                        switch (param.In.Value)
+                        {
+                            case ParameterLocation.Path:
+                                path = path.Replace("{" + param.Name + "}", HttpUtility.UrlEncode(valueStr));
+                                break;
+
+                            case ParameterLocation.Query:
+                                query.Append((query.Length == 0) ? "?" : "&");
+                                query.Append(param.Name);
+                                query.Append('=');
+                                query.Append(HttpUtility.UrlEncode(valueStr));
+                                break;
+
+                            case ParameterLocation.Header:
+                                headers.Add(param.Name, valueStr);
+                                break;                           
+
+                            case ParameterLocation.Cookie:
```

Comment:
>case ParameterLocation.Cookie: [](http://example.com/codeflow?start=28&length=30)

nit ... we don't actually need this, it will be caught by default. 
But it's also a good reminder that we need to impl.

Reply:
I know, I only use that as a reminder and make the code clearer